src/kit.ts

```diff
@@ -292,11 +292,33 @@ export async function kitIfCompiler(bin: string, pr?: ProgressReporter): Promise
         if (version === null) {
             return null;
         }
-        const gxx_fname = fname.replace(/gcc/, 'g++');
-        const gxx_bin = path.join(path.dirname(bin), gxx_fname);
-        const gccCompilers: { [lang: string]: string } = { C: bin };
-        if (await fs.exists(gxx_bin)) {
-            gccCompilers.CXX = gxx_bin;
+        const gccCompilers: { [lang: string]: string } = { };
+        const gxx_fname1 = fname.replace(/gcc/, 'g++');
+        const gxx_bin1 = path.join(path.dirname(bin), gxx_fname1);
+        if (await fs.exists(gxx_bin1)) {
+            // Names like x86_64-pc-linux-gnu-gcc-11.1.0
+            gccCompilers.C = bin;
+            // Names like x86_64-pc-linux-gnu-g++-11.1.0
+            gccCompilers.CXX = gxx_bin1;
+        } else {
+            const fname2 = fname.replace(/gcc(-\d+(\.\d+(\.\d+)?)?)/, 'gcc');
+            const bin2 = path.join(path.dirname(bin), fname2);
+            const gxx_fname2 = fname.replace(/gcc(-\d+(\.\d+(\.\d+)?)?)/, 'g++');
+            const gxx_bin2 = path.join(path.dirname(bin), gxx_fname2);
+            // Ensure the version is match
+            const version2 = await getCompilerVersion('GCC', bin2, pr);
+            const version_is_match = version2 === null ? false : version2.fullVersion === version.fullVersion;
+            if (version_is_match && await fs.exists(bin2)) {
+                // Names like x86_64-pc-linux-gnu-gcc
+                gccCompilers.C = bin2;
+            } else {
+                // Names like x86_64-pc-linux-gnu-gcc-11.1.0
+                gccCompilers.C = bin;
+            }
+            if (version_is_match && await fs.exists(gxx_bin2)) {
```

Comment:
I think this is the issue causing the tests to fail. `fname1 === fname2` because there was no substitution made.  As a result `version2` is the same as `version`.  No substitution is made for `gxx_fname2` for the same reason and `gxx_bin2` exists because it's not `g++`.

In this block you should make sure that a substitution was made before marking a C compiler as a CXX compiler.  You'll need to do the same in the clang code.

Reply:
Address the comments.